### PR TITLE
Migrate `ReportsScreen` to a fragment

### PR DESCRIPTION
**Pull request by msasikanth**
Viewbinding migration: https://app.clubhouse.io/simpledotorg/story/1828/migrate-app-to-use-view-binding-instead-of-kotlin-synthetic-view-accessors

### CODE REVIEW

**Comment by msasikanth** [comment]
Blocked by #2533

File: app/src/main/java/org/simple/clinic/home/report/ReportsScreen.kt [hunk 1]
```diff
@@ -45,6 +47,23 @@ class ReportsScreen : BaseScreen<
...
 
+  override fun defaultModel() = ReportsModel.create()
+
+  override fun bindView(layoutInflater: LayoutInflater, container: ViewGroup?) =
+      ScreenReportBinding.inflate(layoutInflater, container, false)
+
+  override fun uiRenderer() = ReportsUiRenderer(this)
+
+  override fun events() = Observable
+      .mergeArray(webViewBackClicks)
```
Comment (vinaysshenoy) [inline app/src/main/java/org/simple/clinic/home/report/ReportsScreen.kt:46]: Can we also report events to analytics here? I think we might have forgotten to do that in the original screen implementation?

**Review by vinaysshenoy — CHANGES_REQUESTED** [review]
Overall PR looks good, needs a small change and we should be good to go.

File: app/src/main/java/org/simple/clinic/home/report/ReportsEvent.kt [hunk 2]
```diff
@@ -1,8 +1,9 @@
 package org.simple.clinic.home.report
 
 import org.simple.clinic.util.Optional
+import org.simple.clinic.widgets.UiEvent
 
-sealed class ReportsEvent
+sealed class ReportsEvent : UiEvent
```
Comment (vinaysshenoy) [inline app/src/main/java/org/simple/clinic/home/report/ReportsEvent.kt:6]: We'll also need to override `analyticsName` on the events themselves in order to actually report them.

Comment (msasikanth) [reply app/src/main/java/org/simple/clinic/home/report/ReportsEvent.kt:6]: I completely forgot about that 🤦🏾

File: app/src/main/java/org/simple/clinic/home/report/ReportsEvent.kt [hunk 3]
```diff
@@ -1,9 +1,13 @@
...
 
 data class ReportsLoaded(val reportsContent: Optional<String>) : ReportsEvent()
 
-object WebBackClicked : ReportsEvent()
+object WebBackClicked : ReportsEvent() {
+
+  override val analyticsName = "Reports Back clicked"
```
Comment (vinaysshenoy) [inline app/src/main/java/org/simple/clinic/home/report/ReportsEvent.kt:12]: Can we follow the same convention as the rest of the screens: "<Screen Name>:<Event name>" please?

Comment (vinaysshenoy) [inline app/src/main/java/org/simple/clinic/home/report/ReportsEvent.kt:12]: Can we follow the same convention as the other screen events please?

`<Screen Name>:<Event name>`

Comment (msasikanth) [reply app/src/main/java/org/simple/clinic/home/report/ReportsEvent.kt]: Done ✅

**Review by vinaysshenoy — APPROVED** [review]

**Pull request merged**